### PR TITLE
Ads from Bitfly Ads Server

### PR DESCRIPTION
**Pull request by manuelsc**

### CODE REVIEW

File: src/app/components/ad/ad.component.scss [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2021 Bitfly GmbH
```
Comment (D13ce) [inline src/app/components/ad/ad.component.scss:1]: Minor: We might want to update this to 2023 in all files. Thoughts?

File: src/app/components/ad/ad.component.spec.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing'
 import { IonicModule } from '@ionic/angular'
 
-import { TextAdComponent } from './text-ad.component'
+import { AdComponent } from './ad.component'
 
 describe('TextAdComponent', () => {
```
Comment (D13ce) [inline src/app/components/ad/ad.component.spec.ts:6]: Shouldn't this be adapted too?

File: src/app/components/ad/ad.component.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		private sanitizer: DomSanitizer,
+		private modalController: ModalController,
+		private merchantUtils: MerchantUtils
+	) {}
+
+	ngOnInit() {
+		this.merchantUtils.hasAdFree().then((isAdFree) => {
+			this.isAdFree = isAdFree
+		})
+		this.adUtils.get(this.location).then((data) => {
```
Comment (D13ce) [inline src/app/components/ad/ad.component.ts:34]: Minimal: You could skip this call for when `this.isAdfree == true`.
I know that this is checked within `adUtils.get` too, but checking it here would be more performant.

Feel free to ignore and to resolve this conversation.

**Review by D13ce — DISMISSED** [review]
lgtm

Spotted three minor things I'd like to know your opinion on before merging.

File: src/app/components/ad/ad.component.spec.ts (hunk 2 above)
Comment (manuelsc) [reply src/app/components/ad/ad.component.spec.ts]: Thanks, fixed

File: src/app/components/ad/ad.component.scss (hunk 1 above)
Comment (manuelsc) [reply src/app/components/ad/ad.component.scss:1]: Don't have a strong opinion on this. Best to update all at once in a separate PR though.

File: src/app/components/ad/ad.component.ts (hunk 3 above)
Comment (manuelsc) [reply src/app/components/ad/ad.component.ts:34]: The first thing hasAdFree does (or to be more precise the underlying getCurrentPlanConfirmed method) is checking if the user is signed in and if not returning the default package. So there isn't really any performance gain by checking here as well.

**Review by D13ce — APPROVED** [review]
lgtm

**Pull request merged**